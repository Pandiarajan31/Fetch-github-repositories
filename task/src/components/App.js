import ApolloClient, { InMemoryCache } from 'apollo-boost';
import { ApolloProvider } from 'react-apollo';
import React, { Component } from 'react';
import Radium from 'radium';
// import history from 'browserHistory';
import createBrowserHistory from 'history/createBrowserHistory';

import { styles } from './styles/styles';
import Dashboard from './utility/dashboard';
import { Router, Route, Switch,Redirect } from 'react-router-dom';
import Login from "./utility/Login"
const client = new ApolloClient({
  uri: 'https://api.github.com/graphql',
  cache: new InMemoryCache(),
  request: operation => {
    operation.setContext({
      headers: {
        authorization: `Bearer ${localStorage.getItem("access_token")}`
      }
    });
  }
});
const browserHistory = createBrowserHistory();

class App extends Component {

  render() {
    return (
      <ApolloProvider client={ client } >
                <Router history={browserHistory}>
                <Switch>

        <div className='App'
          style={ styles.app }>
          <header className='App-header'
            style={ styles.header }>
            <h1>Fetch GitHub Details</h1>
          </header>
          <Route exact path='/' component={props => <Login {...props} />} />
          <Route path='/dashboard' component={props => (localStorage.getItem("access_token") !== null || "") ? <Dashboard {...props} />:<Redirect to='/' />} />

          <div style={ styles.space }></div>
        </div>
        </Switch>

        </Router>

      </ApolloProvider>
    );
  }
}

export default Radium(App);
