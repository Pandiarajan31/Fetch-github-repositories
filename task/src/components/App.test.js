import React from 'react';
import { StyleRoot } from 'radium';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <StyleRoot>
      <App />
    </StyleRoot>,
  div
);
  ReactDOM.unmountComponentAtNode(div);
});


test("snapshot testing",()=>{
  const component=renderer.create(
    <StyleRoot>
      <App />
    </StyleRoot>
  );
  let tree=component.toJSON();
  expect(tree).toMatchSnapshot();

})

