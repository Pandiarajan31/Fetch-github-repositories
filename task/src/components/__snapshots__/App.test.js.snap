// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
<div
  data-radium={true}
>
  <div
    className="App"
    data-radium={true}
    style={
      Object {
        "height": "110%",
      }
    }
  >
    <header
      className="App-header"
      data-radium={true}
      style={
        Object {
          "color": "#ab20c8",
          "fontSize": "2em",
          "fontWeight": "bolder",
          "padding": "1em 0",
          "textAlign": "center",
        }
      }
    >
      <h1>
        Fetch GitHub Details
      </h1>
    </header>
    <div
      data-radium={true}
    >
      <div
        data-radium={true}
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <input
          className="rmq-2359131e rmq-2437c991 rmq-e28b15b2 search"
          data-radium={true}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Search GitHub by Username"
          required={true}
          style={
            Object {
              "border": "2px solid #ABABAB",
              "color": " #4da6ff",
              "textAlign": "center",
              "width": "400px",
            }
          }
          type="text"
        />
      </div>
      <div>
        <div
          className="repo-list"
          data-radium={true}
        >
          <div
            data-radium={true}
            style={
              Object {
                "marginTop": 10,
                "textAlign": "center",
              }
            }
          >
            <img
              alt="loader"
              src="loading.gif"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      data-radium={true}
      style={
        Object {
          "padding": "20px 0",
        }
      }
    />
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "@media (max-width: 415px){ .rmq-2359131e{width: 90% !important;
font-size: .8em !important;}}
@media (max-width: 315px){ .rmq-2437c991{font-size: .6em !important;}}
@media (max-width: 250px){ .rmq-e28b15b2{font-size: .5em !important;}}",
      }
    }
  />
</div>
`;
